[iterator.concept.winc] Move concept definition near its description

jensmaurer opened the pull request:
Also rename the placeholder type I to T, used in the specification
of the exposition-only concepts is-integer-like and
is-signed-integer-like, to avoid a presentation conflict
with the placeholder type I referring to an iterator type.

Fixes #4472

jensmaurer commented:
@tkoeppe , rebased and force-pushed.

tkoeppe commented on source/iterators.tex:
```diff
@@ -1312,20 +1312,10 @@
-      { ++i } -> @\libconcept{same_as}@<I&>;   // not required to be equality-preserving
-      i++;                      // not required to be equality-preserving
-    };
 \end{codeblock}
```
Hm, it's a bit weird to have these "bare" codeblocks, which in reality are stuck to their preceding paragraph. The status quo was sort of OK, but the new codeblock you're now introducing below seems to be stuck to the wrong paragraph now. We're lacking a mechanism to introduce something with an unnumbered block at the beginning. We have that for synopses and for grammar, and I suppose for itemdecls, but this use case is none of these. Thoughts?

CaseyCarter replied:
FWIW, I wouldn't mind if all of these concept definitions were changed to use `itemdecl` instead of `codeblock`. I'm not sure how we landed on `codeblock` in the first place; I suspect it was a typo on my part the first time that was then replicated a couple of dozen times.

jensmaurer replied:
Rebased. I've also moved the `\pnum` to before the codeblock.

tkoeppe replied:
I've been adding subclause structures whenever we had more such "bare" codeblocks (of which recent papers added a lot), but maybe itemdecls are the way to go here. In any case, I'm afraid this needs another rebase.

tkoeppe replied:
@jensmaurer would you like to take another stab at this?

jensmaurer committed:
[iterator.concept.winc] Rename placeholder type I to T

These types are used in the specification of the exposition-only
concepts is-integer-like and is-signed-integer-like.  The rename
avoids a presentation conflict with the placeholder type I
referring to an iterator type.

jensmaurer committed:
[iterator.concept.winc] Move definition of concept weakly_incrementable
